rest/model/admin.go

```diff
@@ -1536,33 +1536,54 @@ func (j *APIJIRANotificationsConfig) BuildFromService(h interface{}) error {
 	case evergreen.JIRANotificationsConfig:
 		config = &v
 	default:
-		return errors.Errorf("expected *evergreen.APIJIRANotificationsConfig, but got %T instead", h)
+		return errors.Errorf("expected *evergreen.JIRANotificationsConfig, but got %T instead", h)
 	}
 
-	if len(config.CustomFields) == 0 {
-		return nil
-	}
+	j.CustomFields = make(map[string]map[string]string)
+	for _, project := range config.CustomFields {
+		apiProject := make(map[string]string)
+		for _, field := range project.Fields {
+			apiProject[field.Field] = field.Template
+		}
 
-	m, err := config.CustomFields.ToMap()
-	if err != nil {
-		return errors.Wrap(err, "failed to build jira custom field configuration")
-	}
+		components := strings.Join(project.Components, ",")
+		if len(components) > 0 {
+			apiProject["components"] = components
```

Comment:
Instead of doing this string matching, why not have discrete fields for components and labels?

Reply:
done

Reply:
![Screen Shot 2019-10-31 at 12 00 26 PM](https://user-images.githubusercontent.com/17027265/67964092-0d867700-fbd6-11e9-9edc-5337e2b56720.png)